src/view/maintoolbar.cpp

```diff
@@ -360,3 +360,14 @@ void MainToolBar::onMinimize(bool /*checked*/)
 	auto *mainWindow = MainWindow::find(parentWidget());
 	mainWindow->minimize();
 }
+
+void MainToolBar::onClose(bool /*checked*/)
```

Comment:
This only works with the application title bar right? Maybe some handling in the main window itself would be required to make it work with the system title bar?

Reply:
This is true.
In order to make it work the same way when it's closed by the system title bar, I've added some logic to `MainWindow::closeEvent`.
This function also catches every other method of closing like `right-click->close` or `alt+f4`, or any other thing available, and they're not distinguishable.
I thought this could be a problem; however, I've checked that other apps that use system tray work the same way, so I expect that It shouldn't be a problem. At least it's not uncommon.

Now, when you select the option to close to system tray, all these methods will result in `spotify-qt` staying active in tray. In order to quit, one shall right-click the tray icon and select Quit, or use Ctrl+Q inside the app, or use menu->Quit.